core/chain-configs/src/genesis_config.rs

```diff
@@ -454,26 +455,40 @@ pub enum GenesisValidationMode {
 
 impl Genesis {
     pub fn new(config: GenesisConfig, records: GenesisRecords) -> Self {
-        Self::new_validated(config, records, GenesisValidationMode::Full)
+        Self::new_validated(config, records, GenesisValidationMode::Full).unwrap()
     }
 
     pub fn new_with_path<P: AsRef<Path>>(config: GenesisConfig, records_file: P) -> Self {
-        Self::new_with_path_validated(config, records_file, GenesisValidationMode::Full)
+        Self::new_with_path_validated(config, records_file, GenesisValidationMode::Full).unwrap()
     }
 
-    /// Reads Genesis from a single file.
-    /// the file can be JSON with comments
-    pub fn from_file<P: AsRef<Path>>(path: P, genesis_validation: GenesisValidationMode) -> Self {
-        let mut file = File::open(path).expect("Could not open genesis config file.");
+    /// Reads Genesis from a single JSON file, the file can be JSON with comments
+    /// This function will collect all errors regarding genesis.json and push them to validation_errors
+    pub fn from_file<P: AsRef<Path>>(
+        path: P,
+        genesis_validation: GenesisValidationMode,
+    ) -> Result<Self, ValidationError> {
+        let mut file = File::open(path).map_err(|_| ValidationError::GenesisFileError {
```

Comment:
this is the current:
`#[derive(thiserror::Error, Debug)]
pub enum ValidationError {
    #[error("config.json semantic issue: {error_message}")]
    ConfigSemanticsError { error_message: String },
    #[error("genesis.json semantic issue: {error_message}")]
    GenesisSemanticsError { error_message: String },
    #[error("config.json file issue: {error_message}")]
    ConfigFileError { error_message: String },
    #[error("genesis.json file issue: {error_message}")]
    GenesisFileError { error_message: String },
    #[error("node_key.json file issue: {error_message}")]
    NodeKeyFileError { error_message: String },
    #[error("validator_key.json file issue: {error_message}")]
    ValidatorKeyFileError { error_message: String },
    #[error("cross config files semantic issue: {error_message}")]
    CrossFileSematicError { error_message: String },
}`

How about I change it to 
`pub enum ReadConfigError {
    #[error("config.json validation issue: {error_message}")]
    ConfigValidationError { error_message: String },
    #[error("genesis.json validation issue: {error_message}")]
    GenesisValidationError { error_message: String },
    #[error("config.json file issue: {error_message}")]
    ConfigFileError { error_message: String },
    #[error("genesis.json file issue: {error_message}")]
    GenesisFileError { error_message: String },
    #[error("node_key.json file issue: {error_message}")]
    NodeKeyFileError { error_message: String },
    #[error("validator_key.json file issue: {error_message}")]
    ValidatorKeyFileError { error_message: String },
    #[error("cross config files validation issue: {error_message}")]
    CrossFileValidationError { error_message: String },
}`

and change
`pub struct ValidationErrors(Vec<ValidationError>);`
to 
`pub struct ReadConfigErrors(Vec<ReadConfigError>);`

This way the name makes more sense, also I get to have all errrors with the same Enum type so that it's easier to push them into the same `ReadConfigErrors` object to report all errors together.

Reply:
I'll leave this as is since all these errors are identified while doing config validation.